pywhy_stats/api.py

```diff
@@ -59,32 +60,37 @@ def independence_test(
 
     See Also
     --------
-    fisherz : Fisher's Z test for independence
-    kci : Kernel Conditional Independence test
+    pywhy_stats.independence.fisherz : Fisher's Z test for independence
+    pywhy_stats.independence.kci : Kernel Conditional Independence test
     """
     method_module: ModuleType
     if method == Methods.AUTO:
         method_module = Methods.KCI
+    elif not isinstance(method, Methods):
+        raise ValueError(
+            f"Invalid method type. Expected one of {Methods.__members__.keys()}, "
+            f"but got {method}."
+        )
     else:
-        method_module = method
+        method_module = method  # type: ignore
 
     if method_module == Methods.FISHERZ:
         if condition_on is None:
             data = [X, Y]
         else:
             data = [X, Y, condition_on]
         for _data in data:
-            _, pval = scipy.stats.normaltest(_data)
+            res = scipy.stats.normaltest(_data, axis=0)
 
             # XXX: we should add pinguoin as an optional dependency for doing multi-comp stuff
-            if pval < 0.05:
+            if np.atleast_1d(res.pvalue).any() < 0.05:
```

Comment:
Don't get this line, are there multiple p-values? I thought the scipy test only returns one (with the test statistic).

Reply:
Yes if the columns are multivariate, then it is an array of pvalues: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.normaltest.html